MekHQ/src/mekhq/gui/dialog/PartsStoreDialog.java

```diff
@@ -377,181 +386,381 @@ public void filterParts() {
+                        break;
+                    }
+                }
+                if (null == contract) {
+                    MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                            "AtB: used bonus part but no contract has bonus parts available."); //$NON-NLS-1$
+                } else {
+                    contract.useBonusPart();
+                }
+            }
+        } else if(purchase) {
+            campaign.getShoppingList().addShoppingItem(part.getAcquisitionWork(), quantity, campaign);
+        } else {
+            while(quantity > 0) {
+                campaign.addPart(part.clone(), 0);
+                quantity--;
+            }
+        }
     }
 
     private void setSelectedPart() {
-    	int row = partsTable.getSelectedRow();
-		if(row < 0) {
-			return;
-		}
-		selectedPart = partsModel.getPartAt(partsTable.convertRowIndexToModel(row));
+        int row = partsTable.getSelectedRow();
+        if(row < 0) {
+            return;
+        }
+        selectedPart = partsModel.getPartAt(partsTable.convertRowIndexToModel(row));
     }
 
     public Part getPart() {
-    	return selectedPart;
+        return selectedPart;
     }
 
     public static String getPartsGroupName(int group) {
-    	switch(group) {
-    	case SG_ALL:
-    		return "All Parts";
-    	case SG_ARMOR:
-    		return "Armor";
-    	case SG_SYSTEM:
-    		return "System Components";
-    	case SG_EQUIP:
-    		return "Equipment";
-    	case SG_LOC:
-    		return "Locations";
-    	case SG_WEAP:
-    		return "Weapons";
-    	case SG_AMMO:
-    		return "Ammunition";
-    	case SG_MISC:
-    		return "Miscellaneous Equipment";
-    	case SG_ENGINE:
-    		return "Engines";
-    	case SG_GYRO:
-    		return "Gyros";
-    	case SG_ACT:
-    		return "Actuators";
-    	case SG_COCKPIT:
-    		return "Cockpits";
-    	case SG_BA_SUIT:
-    		return "Battle Armor Suits";
-    	case SG_OMNI_POD:
-    	    return "Empty OmniPods";
-    	default:
-    		return "?";
-    	}
+        switch(group) {
+        case SG_ALL:
+            return "All Parts";
+        case SG_ARMOR:
+            return "Armor";
+        case SG_SYSTEM:
+            return "System Components";
+        case SG_EQUIP:
+            return "Equipment";
+        case SG_LOC:
+            return "Locations";
+        case SG_WEAP:
+            return "Weapons";
+        case SG_AMMO:
+            return "Ammunition";
+        case SG_MISC:
+            return "Miscellaneous Equipment";
+        case SG_ENGINE:
+            return "Engines";
+        case SG_GYRO:
+            return "Gyros";
+        case SG_ACT:
+            return "Actuators";
+        case SG_COCKPIT:
+            return "Cockpits";
+        case SG_BA_SUIT:
+            return "Battle Armor Suits";
+        case SG_OMNI_POD:
+            return "Empty OmniPods";
+        default:
+            return "?";
+        }
+    }
+
+    private Person getLogisticsPerson() {
```

Comment:
Does this get re-initialized every time the window is re-opened? I'm perfectly happy with a cached approach, but it may run into issues if you open the parts window, close it, then your logistics person changes, then you open it back up again.

Reply:
Yes, these dialog instances are not cached or stored.